test/functional/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyTest.java

```diff
@@ -370,4 +370,36 @@ public void test_boostrap_return_constant_MethodType() {
 		AssertJUnit.assertEquals(expected, mt);
 		AssertJUnit.assertTrue(expected == mt);
 	}
+
+	// test that if resolved CallSite is null, the same error is rethrown
+	@Test(groups = { "level.extended" })
+	public void test_CallSiteNullErrorRethrown () {
+		// BSME is expected on the first run
+		boolean bootstrapMethodErrorThrown1 = false;
+		try {
+			com.ibm.j9.jsr292.indyn.GenIndyn.test_CallSiteNullErrorRethrown();
+		} catch ( java.lang.BootstrapMethodError e ) {
+			bootstrapMethodErrorThrown1 = true;
+		}
+
+		if ( bootstrapMethodErrorThrown1 == false ) {
+			Assert.fail( "BootstrapMethodError was expected for null CallSite" );
+		}
+
+		// The same BSME is expected on the second run
+		boolean bootstrapMethodErrorThrown2 = false;
+		try {
+			com.ibm.j9.jsr292.indyn.GenIndyn.test_CallSiteNullErrorRethrown();
+		} catch ( java.lang.BootstrapMethodError e ) {
```

Comment:
I would expect an NPE on java 8 for the first call and the second to succeed.  Is that not the case?

Reply:
yes you are right! I had verified this test with the RI only. The RI does not through an error while OpenJ9 throws an NPE.

Reply:
any objections to matching that?

Reply:
I want J9 to continue to throw the NPE for older versions.  Otherwise, we'd have some serious issues trying to use null as an object in code that expects it to never be null.